ERCS/erc-7818.md

```diff
@@ -0,0 +1,162 @@
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-11-13
+requires: 20
+---
+
+## Abstract
+
+Introduces an extension for [ERC-20](./eip-20.md) tokens, which facilitates the implementation of an expiration mechanism. Through this extension, tokens have a predetermined validity period, after which they become invalid and can no longer be transferred or used. This functionality proves beneficial in scenarios such as time-limited bonds, loyalty rewards, or game tokens necessitating automatic invalidation after a specific duration. The extension is crafted to seamlessly align with the existing [ERC-20](./eip-20.md) standard, ensuring smooth integration with the prevailing token smart contract while introducing the capability to govern and enforce token expiration at the contract level.
+
+## Motivation
+
+This extension facilitates the development of [ERC-20](./eip-20.md) standard compatible tokens featuring expiration dates. This capability broadens the scope of potential applications, particularly those involving time-sensitive assets. Expirable tokens are well-suited for scenarios necessitating temporary validity, including
+
+- Bonds or financial instruments with defined maturity dates
+- Time-constrained assets within gaming ecosystems
+- Next-gen loyalty programs incorporating expiring rewards or points
+- Prepaid credits for utilities or services (e.g., cashback, data packages, fuel, computing resources) that expire if not used within a specified time frame
+- Postpaid telecom data package allocations that expire at the end of the billing cycle, motivating users to utilize their data before it resets
+- Tokenized e-Money for a closed-loop ecosystem, such as transportation, food court, and retail payments
+
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Implementers of this **REQUIRED** inheritance [ERC-20](./eip-20.md) interface and **MUST** have all of the following functions and all function behavior **MUST** meet the specification.
+
+```solidity
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity >=0.8.0 <0.9.0;
+
+/**
+ * @title ERC-7818: Expirable ERC20
+ * @dev Interface for creating expirable ERC20 tokens.
+ */
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+interface IERC7818 is IERC20 {
+
+    /**
+     * @dev Retrieves the balance of a specific `identifier` owned by an account.
+     * @param account The address of the account.
+     * @param identifier The Identifier "MAY" represents an epoch, round, period, or token identifier.
+     * @return uint256 The balance of the specified `identifier`.
+     * @notice `identifier` "MUST" represent a unique identifier, and its meaning "SHOULD"
+     * align with how contract maintain the `epoch` in the implementing contract.
```

Comment:
Putting requirements in the comments is allowed, but I find it a bit more difficult to read. You might want to move them outside the code fence? This applies to all the netspec comments.